test/cases/coerced_tests.rb

```diff
@@ -1130,7 +1130,7 @@ def test_query_cached_even_when_types_are_reset_coerced
       Task.find(1)
 
       # Preload the type cache again (so we don't have those queries issued during our assertions)
-      Task.connection.send(:reload_type_map)
+      Task.connection.send(:reload_type_map) if Task.connection.respond_to?(:reload_type_map, true)
```

Comment:
Can `Task.connection.send(:reload_type_map)` not simply be removed since `respond_to?` will always be `false` for the SQL Server adapter in Rails 7?

Reply:
It's true that the whole line can be removed. I choose to add the `respond_to?` check to honor the description of the test `Same as original test except that we expect one query to be performed...`. This way the body of the test is Rails 7 + the extra assertion. Easier to maintain in my opinion. What do you think?

Reply:
Removed the call and changed the test comment